server/controllers/recommendation/bookToPpudController.ts

```diff
@@ -128,7 +140,7 @@ async function post(req: Request, res: Response, _: NextFunction) {
       dateOfSentence: nomisOffence.sentenceDate,
       licenceExpiryDate: nomisOffence.licenceExpiryDate,
       releaseDate: nomisOffence.releaseDate,
-      sentenceLength: { partDays: offenceTerm.days, partMonths: offenceTerm.months, partYears: offenceTerm.years },
+      sentenceLength: { partDays: offenceTerm?.days, partMonths: offenceTerm?.months, partYears: offenceTerm?.years },
```

Comment:
This will fail when it gets to PPUD Automation if offenceTerm is null.  If offence term is null, sentenceLength can be null, but if sentenceLength is not null then it must have values.  The values can be 0.

Reply:
fixed